tests/bluetooth/mesh/brg/src/main.c

```diff
@@ -0,0 +1,430 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/ztest.h>
+#include <zephyr/net/buf.h>
+#include <zephyr/bluetooth/mesh.h>
+#include <stdlib.h>
+
+#include "settings.h"
+#include "brg_cfg.h"
+
+#define TEST_VECT_SZ (CONFIG_BT_MESH_BRG_TABLE_ITEMS_MAX + 1)
+
+static struct test_brg_cfg_row {
+	uint8_t direction;
+	uint16_t net_idx1;
+	uint16_t net_idx2;
+	uint16_t addr1;
+	uint16_t addr2;
+} test_vector[TEST_VECT_SZ];
+
+#define ADDR1_BASE (1)
+#define ADDR2_BASE (100)
+
+/**** Helper functions ****/
+static void setup(void *f)
+{
+	/* create test vector */
+	for (int i = 0; i < TEST_VECT_SZ; i++) {
+		test_vector[i].direction = i < (TEST_VECT_SZ / 2) ? 1 : 2;
+		test_vector[i].net_idx1 = (i/8);
+		test_vector[i].addr1 = ADDR1_BASE + i;
+		test_vector[i].net_idx2 = (i/8) + 16;
+		test_vector[i].addr2 = ADDR2_BASE + i;
+	}
+
+}
+
+/**** Mocked functions ****/
+
+void bt_mesh_settings_store_schedule(enum bt_mesh_settings_flag flag)
+{
+	ztest_check_expected_value(flag);
+}
+
+int settings_save_one(const char *name, const void *value, size_t val_len)
+{
+	ztest_check_expected_data(name, strlen(name));
+	ztest_check_expected_value(val_len);
+	ztest_check_expected_data(value, val_len);
+	return 0;
+}
+
+int settings_delete(const char *name)
+{
+	ztest_check_expected_data(name, strlen(name));
+	return 0;
+}
+
+static void check_fill_all_bt_entries(void)
+{
+	int err;
+
+	for (int i = 0; i < TEST_VECT_SZ; i++) {
+
+		if (i < CONFIG_BT_MESH_BRG_TABLE_ITEMS_MAX) {
+			ztest_expect_value(bt_mesh_settings_store_schedule, flag,
+				BT_MESH_SETTINGS_BRG_PENDING);
+		}
+
+		err = brg_cfg_tbl_add(test_vector[i].direction, test_vector[i].net_idx1,
+			test_vector[i].net_idx2, test_vector[i].addr1, test_vector[i].addr2);
+
+		if (i != CONFIG_BT_MESH_BRG_TABLE_ITEMS_MAX) {
+			zassert_equal(err, 0);
+		} else {
+			zassert_equal(err, -ENOMEM);
+		}
+	}
+}
+
+static void check_delete_all_bt_entries(void)
+{
+	int err;
+
+	for (int i = 0; i < TEST_VECT_SZ; i++) {
+
+		if (i < CONFIG_BT_MESH_BRG_TABLE_ITEMS_MAX) {
+			ztest_expect_value(bt_mesh_settings_store_schedule, flag,
+				BT_MESH_SETTINGS_BRG_PENDING);
+		}
+
+		err = brg_cfg_tbl_remove(test_vector[i].net_idx1, test_vector[i].net_idx2,
+			test_vector[i].addr1, test_vector[i].addr2);
+		zassert_equal(err, 0);
+	}
+}
+
+
+static void check_brg_cfg_tbl_reset(void)
+{
+	int err;
+
+	ztest_expect_data(settings_delete, name, "bt/mesh/brgen");
+	ztest_expect_data(settings_delete, name, "bt/mesh/brgbt");
+	err = brg_cfg_tbl_reset();
+	zassert_equal(err, 0);
+}
+
+/**** Tests ****/
+
+ZTEST_SUITE(bt_mesh_brg_cfg, NULL, NULL, setup, NULL, NULL);
+
+ZTEST(bt_mesh_brg_cfg, test_basic_functionality_storage)
```

Comment:
Please add test scenario description in the front of each test case like here: https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/bluetooth/mesh/delayable_msg/src/main.c#L266-L270
It will help to maintain them.

Reply:
it isn't resolved